src/app/_components/Share/index.tsx

```diff
@@ -0,0 +1,70 @@
+'use client'
+
+import { useEffect, useState } from 'react'
+import { FacebookShareButton, LinkedinShareButton, TwitterShareButton } from 'react-share'
+
+import styles from './styles.module.css'
+
+import { LinkIcon, ShareIcon } from '@/app/_icons/icons'
+import { FacebookIcon, LinkedInIcon, TwitterIcon } from '@/app/_icons/socialIcons'
+
+export default function Share() {
+  const copyURLToClipboard = () => {
+    navigator.clipboard.writeText(document.URL)
+  }
+
+  const shareTo = async () => {
+    await navigator.share({
+      title: 'cenas',
+      text: 'mais cenas',
+      url: window.location.href,
+    })
+  }
+
+  const [url, setUrl] = useState('')
```

Comment:
Can't we use `window.location.href` directly, instead of setting url in a useEffect?

Reply:
nvm, it's because of nextjs shenanigans, of course

Reply:
yep, classic nextjs shenanigans 😅 
for future refference: for this type of projects would astro (or similar) be leaner / less bloated code-wise than nextjs?